gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineExactDependencies.java

```diff
@@ -89,15 +89,13 @@ private static void configureSourceSet(
             TaskProvider<CheckImplicitDependenciesParentTask> checkImplicitDependencies) {
         Configuration implementation =
                 project.getConfigurations().getByName(sourceSet.getImplementationConfigurationName());
-        Configuration compile = project.getConfigurations().getByName(sourceSet.getCompileConfigurationName());
```

Comment:
This break is hard to work around. Ideally we would still get the compile configuration as an optional if present. However with Gradle 7, the `getCompileConfigurationName` method no longer exists and the method for deriving the configuration name based on the task is private ([code ref](https://github.com/gradle/gradle/blob/a2ea10896cbdca85a4fb7b0e30d638f57dbe1533/subprojects/plugins/src/main/java/org/gradle/api/internal/tasks/DefaultSourceSet.java#L143-L145)).

Reply:
Can we just reimplement that logic while we support Gradle 6?

Reply:
Yep, its actually quite the simple method. Added it for now.